### PR TITLE
Adapt to transformers 4.44

### PR DESCRIPTION
**Pull request by Jadon-Chan**

### CODE REVIEW

File: MoA/attention/cache_utils.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ def __init__(
         band_size: List[List[int]],
         pattern_index: List[List[int]],
     ) -> None:
-        self.seen_tokens = 0
+        self.cache_position = 0
```
Comment (fuvty) [inline MoA/attention/cache_utils.py:17]: Is this change due to the interface change in Transformers? If so, it's perfectly fine.

File: MoA/attention/cache_utils.py [hunk 2]
```diff
@@ -676,6 +677,7 @@ def __init__(
             update_cache_content (*optional*, `bool`):
                 Whether to update the cache content when calling `.update`. If not provided, it will be set to `True`. Set to False if the kernel update the content
         """
+        super().__init__()
```
Comment (fuvty) [inline MoA/attention/cache_utils.py:680]: Init the original cache is necessary right?

File: MoA/kernels/mixture_of_attention.py
Comment (fuvty) [inline MoA/kernels/mixture_of_attention.py:1]: Please kindly revert this change and ensure the retrieval accuracy still stay the same.

File: MoA/models/llama/modeling_llama.py [hunk 3]
```diff
@@ -54,33 +54,24 @@ def LlamaModel_MixtureAttention_forward(
     input_ids: torch.LongTensor = None,
     attention_mask: Optional[torch.Tensor] = None,
     position_ids: Optional[torch.LongTensor] = None,
-    past_key_values: Optional[List[torch.FloatTensor]] = None,
+    past_key_values: Optional[Union[Cache, List[torch.FloatTensor]]] = None,
```
Comment (fuvty) [inline MoA/models/llama/modeling_llama.py:57]: Perhaps Cache should be replaced as StaticCircularCache? Since this is the forward function of MoA and the default cache is not the regular cache. We may want to highlight that.

File: MoA/models/llama/modeling_llama.py [hunk 4]
```diff
@@ -219,14 +207,15 @@ class LlamaMixtureAttention(LlamaAttention):
     def forward(
         self,
         hidden_states: torch.Tensor,
-        attention_mask: Optional[torch.LongTensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
         position_ids: Optional[torch.LongTensor] = None,
-        past_key_value: Optional[StaticCircularCache] = None,
+        past_key_value: Optional[Cache] = None,
```
Comment (fuvty) [inline MoA/models/llama/modeling_llama.py:212]: I think it should be `StaticCircularCache` and `LongTensor`. It would be best if you could help us double-check

**Review by fuvty — CHANGES_REQUESTED** [review]
Thank you so much for the PR! Please kindly see to the comments to improve the PR further :)

File: MoA/attention/cache_utils.py (hunk 1 above)
Comment (Jadon-Chan) [reply MoA/attention/cache_utils.py:17]: Yes. The seen_tokens attribute is deprecated and was removed in v4.41.

File: MoA/attention/cache_utils.py (hunk 2 above)
Comment (Jadon-Chan) [reply MoA/attention/cache_utils.py:680]: Because attribute `_modules` is accessed, we need to run the `__init__` method of parent class (`transformers.cache_utils.Cache` class) to make sure `StaticCircularCache` has this attribute.

File: MoA/models/llama/modeling_llama.py (hunk 3 above)
Comment (Jadon-Chan) [reply MoA/models/llama/modeling_llama.py:57]: You are right. I replace `Cache` with `StaticCircularCache` in next commit.

File: MoA/models/llama/modeling_llama.py (hunk 4 above)
Comment (Jadon-Chan) [reply MoA/models/llama/modeling_llama.py:212]: You are right. I replace Cache with StaticCircularCache in next commit.

**Review by fuvty — APPROVED** [review]
Thank you so much for the prompt reply. All looking good to me!
